pdns/pdns_recursor.cc

```diff
@@ -1253,6 +1253,28 @@ static void startDoResolve(void *p)
     newLat = min(newLat,(uint64_t)(((uint64_t) g_networkTimeoutMsec)*1000)); // outliers of several minutes exist..
     g_stats.avgLatencyUsec=(1-1.0/g_latencyStatSize)*g_stats.avgLatencyUsec + (float)newLat/g_latencyStatSize;
     // no worries, we do this for packet cache hits elsewhere
+
+    auto ourtime = 1000.0*spent-sr.d_totUsec/1000.0; // in msec
```

Comment:
It looks like we are not updating `SyncRes::d_totUsec` when we forward a cache-only (`RD`=0) query to a hard-coded server in `SyncRes::doResolve()`. This would look like we spent a lot of time in the recursor when we didn't.

Reply:
nice! solved.